lib/features/home/presentation/screens/home_screen.dart

```diff
@@ -0,0 +1,245 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+
+import '../../domain/entities/event_entity.dart';
+import '../cubit/home_cubit.dart';
+import '../widgets/event_large_cell.dart';
+import '../widgets/event_list_cell.dart';
+
+class HomeScreen extends StatefulWidget {
+  const HomeScreen({Key? key}) : super(key: key);
+
+  @override
+  State<HomeScreen> createState() => _HomeScreenState();
+}
+
+class _HomeScreenState extends State<HomeScreen> {
+  _getHomeData() => context.read<HomeCubit>().getHomeData();
+
+  @override
+  void initState() {
+    super.initState();
+    _getHomeData();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return BlocBuilder<HomeCubit, HomeState>(
+      builder: (context, state) {
+        if (state is HomeFailure) {
+          return Container(
+            color: Colors.white,
+            child: Center(
+              child: TextButton(
+                onPressed: () {
+                  _getHomeData();
+                },
+                child: Text(
+                  state.message,
+                  style: const TextStyle(
+                    color: Colors.black,
+                    fontWeight: FontWeight.bold,
+                    fontSize: 16,
+                  ),
+                ),
+              ),
+            ),
+          );
+        } else if (state is HomeData) {
+          final homeData = state.homeData;
+          return Scaffold(
+            backgroundColor: const Color(0xFFF9F9FA),
+            appBar: _buildEventsAppBar(homeData.profileImage),
+            body: SafeArea(
+              child: ListView(
+                padding:
+                    const EdgeInsets.symmetric(vertical: 16, horizontal: 20),
+                children: [
+                  _buildSearchTextField(),
+                  if (homeData.myEvents?.isNotEmpty ?? false)
+                    ..._buildMyEventsSection(homeData.myEvents!),
+                  if (homeData.suggestedEvents?.isNotEmpty ?? false)
+                    ..._buildSuggestedEventsSection(homeData.suggestedEvents!),
+                ],
+              ),
+            ),
+            bottomNavigationBar: _buildBottomNavigationBar(),
+          );
+        } else {
+          return Container(
+            color: Colors.white,
+            child: const Center(
+              child: CircularProgressIndicator(),
+            ),
+          );
+        }
+      },
+    );
+  }
+
+  PreferredSizeWidget _buildEventsAppBar(String profileImage) {
```

Comment:
prefer to use class widget rather than function widget

Reply:
When I moved this to `StatelessWidget` I get below error

> The argument type 'EventsAppBar' can't be assigned to the parameter type 'PreferredSizeWidget?'.

@hashish3 Any suggestions?